### ISSUE TITLE
Add support for RSCALE (RFC 7529) to JMAP calendar events

### ISSUE THREAD

**Issue by ksmurchison** [jmapio/jmap#24]
Since the JMAP spec is still evolving you probably want to add 'rscale' and 'skip' properties to the 'recurrence' object.

Question: Why not just use the existing jCal format (RFC 7265) rather than something similar but different?

**Comment by neilj** [comment]
Yes, we do need to add these. I'm waiting for us to finish the Event object spec in CalConnect TC-API, since this is based off the JMAP spec, then change the JMAP spec to match. (There aren't many changes so far, mainly just a few property name changes).

Regarding why not the jCal recurrence spec, we discussed this at TC-API and although they're fairly similar:
* The until date format is different; it's much better for this to be consistent with the rest of the spec.
* The names are not consistent with the rest of this spec (capitalisation, abbreviations).
* The jCal representation does not have a unique representation for each rule (e.g. byDate with a single value could be an array or just a number); this makes it harder to compare if two rules are identical.

In general, we concluded it was better to be self-consistent with the rest of the spec. There are not too many advantages to being the same as jCal; this hasn't seen widespread adoption, and Cyrus thought keeping the code separate would actually be easier for when jCal was ripped out sometime in the future.
